setup-local/src/actionInput/inputValidator.js

```diff
@@ -0,0 +1,112 @@
+const core = require('@actions/core');
+const { v4: uuidv4 } = require('uuid');
+const parseArgs = require('minimist');
+const constants = require('../../config/constants');
+
+const {
+  ALLOWED_INPUT_VALUES: {
+    LOCAL_LOG_LEVEL,
+    LOCAL_TESTING,
+    LOCAL_IDENTIFIER_RANDOM,
+  },
+  INPUT,
+  RESTRICTED_LOCAL_ARGS,
+} = constants;
+
+/**
+ * InputValidator performs validation on the input fields of this
+ * action. The fields are parsed and converted into the required format.
+ */
+class InputValidator {
+  /**
+   * Validates the action input 'local-testing' and returns the
+   * parsed value.
+   * Throws error if it's not a valid value
+   * @param {String} inputLocalTesting Action input for 'local-testing'
+   * @returns {String} One of the values from start/stop/false
+   */
+  static validateLocalTesting(inputLocalTesting) {
+    const localTestingLowered = inputLocalTesting.toLowerCase();
+    // eslint-disable-next-line max-len
+    const validValue = Object.values(LOCAL_TESTING).some((allowedValue) => allowedValue === localTestingLowered);
+
+    if (!validValue) {
+      throw Error(`Invalid input for ${INPUT.LOCAL_TESING}. The valid inputs are: ${Object.values(LOCAL_TESTING).join(', ')}. Refer the README for more details`);
+    }
+
+    return localTestingLowered;
+  }
+
+  /**
+   * Validates the action input 'local-logging-level' and returns the
+   * verbosity level of logging.
+   * @param {String} inputLocalLoggingLevel Action input for 'local-logging-level'
+   * @returns {Number} Logging Level (0 - 3)
+   */
+  static validateLocalLoggingLevel(inputLocalLoggingLevel) {
+    if (!inputLocalLoggingLevel) return 0;
```

Comment:
Is this required? plus if we are returning 0 from here then shouldn't we warn for no logs as done below

Reply:
If the user hasn't passed anything... then that's more like a silent(implicit) thing that no logs will be uploaded.
But if the user passes some value and that's not one of the valid/allowed ones, then we log the warning.

Reply:
Resolving this. discussed offline.